pkg/cache/simple.go

```diff
@@ -133,14 +133,19 @@ func respond(watch Watch, snapshot Snapshot, group Key) {
 		}
 
 		// check that every snapshot resource name is present in the request
+		found := false
 		for _, resource := range resources {
 			resourceName := GetResourceName(resource)
-			if _, exists := names[resourceName]; !exists {
-				glog.V(10).Infof("not responding for %s from %q at %q since %q not requested %v",
-					typ.String(), group, version, resourceName, watch.Names)
-				return
+			if _, exists := names[resourceName]; exists {
+				found = true
+				break
```

Comment:
This seems to check that there exists at least on resource in the snapshot that was requested. For example,
if the snapshot contains resources `a` and `b`; and the request asks for `a` and `c` then we return `a` and `b`.
Is this intentional?